mantidproject/mantid
# Adding plotting options to the muon plotting

The new muon plotting (currently only in Elemental analysis) allows the user to have up to 4 subplot and update dynamically. This PR adds some functionality:
1. Adds (most of) the standard python tool bar
2. Allows the user to remove specific lines from the plot

**To test:**
Email me and I will send you some data (will be Wednesday)
1. Open Elemental Analysis (in muon interfaces)
2. Set user directories to include test data 
[fakeElementData.zip](https://github.com/mantidproject/mantid/files/2419137/fakeElementData.zip)

3. Type in the run number, 2699, into the spin box
4. Click `Load`
5. Click GE1 
5. Once the data has loaded you will see the new plotting. 

Things to test
1. The standard python toolbar works as expected (forward and backward have been removed as it would jump between subplots). The green tick is a normal matplotlib tool bar too, it allows users to edit the subplot.
2. `Add lines` will just print to screen and will be implemented in #23546 
3. `remove lines` will create a pop up with all of the plotted lines and tick boxes. If you tick some and click remove it will remove them from the plot (legend won't update if it is present)
4. The remove lines pop up will raise if you click the button again without closing it or pressing remove
5. If you click GE2 in the elemental analysis GUI you will get two plots. Clicking `remove lines` will now create a selection pop up that lists the subplots. Once you have selected one it will go to the remove pop up.
6. Removing all of the lines from a subplot will close it and if it was the last subplot the plotting window will close. 
 


Fixes #23524. 




---
Does not need to be in release notes as it is just for the new elemental analysis GUI for now.
#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
